pkg/services/ngalert/migration/models/models_test.go

```diff
@@ -0,0 +1,149 @@
+			maxLen:   10,
+			input:    []string{strings.Repeat("A", 15), strings.Repeat("A", 10)},
+			expected: []string{strings.Repeat("A", 10), strings.Repeat("A", 7) + " #2"},
+		},
+		{
+			name:            "when deduplicate causes collision, it deduplicates - caseInsensitive",
+			caseInsensitive: true,
+			maxLen:          10,
+			input:           []string{"A", "a", "a #2", "b", "B", "B #2"},
+			expected:        []string{"A", "a #2", "a #2 #2", "b", "B #2", "B #2 #2"},
+		},
+		{
+			name:     "when deduplicate causes collision, it deduplicates - caseSensitive",
+			maxLen:   10,
+			input:    []string{"a", "a", "a #2", "b", "b", "b #2"},
+			expected: []string{"a", "a #2", "a #2 #2", "b", "b #2", "b #2 #2"},
+		},
+		{
+			name:            "when deduplicate causes collision, it finds next available increment - caseInsensitive",
+			caseInsensitive: true,
+			maxLen:          10,
+			input:           []string{"a", "A #2", "a #3", "A #4", "a #5", "A #6", "a #7", "A #8", "a #9", "A #10", "a"},
+			expected:        []string{"a", "A #2", "a #3", "A #4", "a #5", "A #6", "a #7", "A #8", "a #9", "A #10", "a #11"},
+		},
+		{
+			name:     "when deduplicate causes collision, it finds next available increment - caseSensitive",
+			maxLen:   10,
+			input:    []string{"a", "a #2", "a #3", "a #4", "a #5", "a #6", "a #7", "a #8", "a #9", "a #10", "a"},
+			expected: []string{"a", "a #2", "a #3", "a #4", "a #5", "a #6", "a #7", "a #8", "a #9", "a #10", "a #11"},
+		},
+		{
+			name:            "when deduplicate causes collision enough times, it deduplicates with uid - caseInsensitive",
+			caseInsensitive: true,
+			maxLen:          10,
+			input:           []string{"a", "A #2", "a #3", "A #4", "a #5", "A #6", "a #7", "A #8", "a #9", "A #10", "a #11", "A"},
+			expected:        []string{"a", "A #2", "a #3", "A #4", "a #5", "A #6", "a #7", "A #8", "a #9", "A #10", "a #11", "A_uid-1"},
+		},
+		{
+			name:     "when deduplicate causes collision enough times, it deduplicates with uid - caseSensitive",
+			maxLen:   10,
+			input:    []string{"a", "a #2", "a #3", "a #4", "a #5", "a #6", "a #7", "a #8", "a #9", "a #10", "a #11", "a"},
+			expected: []string{"a", "a #2", "a #3", "a #4", "a #5", "a #6", "a #7", "a #8", "a #9", "a #10", "a #11", "a_uid-1"},
+		},
+	}
+
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			inc := 0
+			mockUidGenerator := func() string {
+				inc++
+				return fmt.Sprintf("uid-%d", inc)
+			}
+			dedup := Deduplicator{
+				set:             make(map[string]int),
+				caseInsensitive: tt.caseInsensitive,
+				maxLen:          tt.maxLen,
+				uidGenerator:    mockUidGenerator,
+			}
+			out := make([]string, 0, len(tt.input))
+			for _, in := range tt.input {
+				d, err := dedup.Deduplicate(in)
+				require.NoError(t, err)
+				out = append(out, d)
+			}
+			require.Equal(t, tt.expected, out)
+		})
+	}
+}
+
+func Test_shortUIDCaseInsensitiveConflicts(t *testing.T) {
+	s := Deduplicator{
```

Comment:
I wonder if it can be simplified to just two lines with predefined UIDs like: 
1. "abc"
2. "Abc"

and then just count number of keys when caseInsensitive is true and false